CRM/Member/BAO/MembershipStatus.php

```diff
@@ -263,21 +251,27 @@ public static function getMembershipStatusByDate(
 
     $membershipStatus = CRM_Core_DAO::executeQuery($query);
 
+    $dates = [
+      'start' => ($startDate && $startDate !== 'null') ? date('Ymd', strtotime($startDate)) : '',
+      'end' => ($endDate && $endDate !== 'null') ? date('Ymd', strtotime($endDate)) : '',
+      'join' => ($joinDate && $joinDate !== 'null') ? date('Ymd', strtotime($joinDate)) : '',
+    ];
+
     while ($membershipStatus->fetch()) {
       $startEvent = NULL;
       $endEvent = NULL;
-      foreach ($events as $eve) {
-        foreach ($dates as $dat) {
+      foreach (['start', 'end'] as $eve) {
+        foreach ($dates as $dat => $date) {
           // calculate start-event/date and end-event/date
-          if (($membershipStatus->{$eve . '_event'} == $dat . '_date') &&
-            ${$dat . 'Date'}
+          if (($membershipStatus->{$eve . '_event'} === $dat . '_date') &&
+            $date
           ) {
             if ($membershipStatus->{$eve . '_event_adjust_unit'} &&
               $membershipStatus->{$eve . '_event_adjust_interval'}
             ) {
-              $month = date('m', strtotime(${$dat . 'Date'}));
-              $day = date('d', strtotime(${$dat . 'Date'}));
-              $year = date('Y', strtotime(${$dat . 'Date'}));
```

Comment:
Ooph, huzzah, remove that...

Reply:
Hey !!- that only just went in - you should see what the prior version looked like!

Reply:
Try this ....

```
   foreach ($dates as $dat) {
      if (${$dat . 'Date'} && ${$dat . 'Date'} != "null") {
        ${$dat . 'Date'} = CRM_Utils_Date::customFormat(${$dat . 'Date'}, '%Y%m%d');

        ${$dat . 'Year'} = substr(${$dat . 'Date'}, 0, 4);

        ${$dat . 'Month'} = substr(${$dat . 'Date'}, 4, 2);

        ${$dat . 'Day'} = substr(${$dat . 'Date'}, 6, 2);
      }
      else {
        ${$dat . 'Date'} = '';
      }
    }
```